### PR TITLE
JAMES-2544 AsyncRetry library can be used with a single scheduler thread

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java [hunk 1]
```diff
@@ -81,6 +85,13 @@ public void done(boolean success) {
         this.mailReferenceSerializer = serializer;
         this.mailQueueView = mailQueueView;
         this.dequeueMetric = metricFactory.generate(DEQUEUED_METRIC_NAME_PREFIX + name.asString());
+
+        this.executor = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    @PreDestroy
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java]: Good catch

**Comment by chibenwa** [comment]
:green_apple:

**Comment by chibenwa** [comment]
https://github.com/linagora/james-project/pull/1998 is an alternative

**Comment by rouazana** [comment]
conflicting, can you rebase?

**Comment by mbaechler** [comment]
> conflicting, can you rebase?

Should not be usefull after https://github.com/linagora/james-project/pull/1998 is eventually merged

**Comment by chibenwa** [comment]
> Should not be usefull after #1998 is eventually merged

Yes I am just waiting for the merge to close this

**Comment by mbaechler** [comment]
Flux usage for Dequeue replaced that PR

**Pull request closed without merging**